### ISSUE TITLE
/transaction/material enhancement

### ISSUE THREAD

**Issue by vae520283995** [paritytech/substrate-api-sidecar#476]
I hope that the metadata returned by /transaction/material is the decoded json data, which is the same as /runtime/metadata, otherwise I need to request the network twice to obtain the network information needed to construct the transaction

**Comment by emostov** [comment]
@vae520283995 why do you need the decoded metadata? 
You should be able to use the encoded metadata to construct the transaction with something like txwrapper or just the @polkadot/metadata package in general.

If there is a compelling reason to add this feature we can probably put it behind a query param so it is not a breaking change and the default payload is smaller.

**Comment by vae520283995** [comment]
@emostov Thank you, Because the transaction contains call_index, for example, Balances_transfer is 0500, and Balances_transfer_keep_alive is 0503. I am worried that this value will change according to the metadata update, so I get it from the decoded metadata, so I need the decoded metadata. By the way, this value will be Change?

**Comment by emostov** [comment]
> By the way, this value will be Change?

Yes, in theory any value in the metadata can change so it is important to always use the latest metadata when constructing a transaction.

> Because the transaction contains call_index, for example, Balances_transfer is 0500, and Balances_transfer_keep_alive is 0503. 

The encoded metadata contains this info as well, but you need a scale codec library to decode it, such as txwrapper or polkadot.js. What are you using to create transactions?

**Comment by vae520283995** [comment]
> The encoded metadata contains this info as well, but you need a scale codec library to decode it, such as txwrapper or polkadot.js. What are you using to create transactions?

I didn't use any package, I created the transaction according to the api of substrate-api-sidecar, so I hope the metadata returned by /transaction/material is the decoded data

**Comment by emostov** [comment]
> I didn't use any package, I created the transaction according to the api of substrate-api-sidecar, so I hope the metadata returned by /transaction/material is the decoded data

Can you clarify why you need the decoded metadata? All the tooling I know of to make transactions can use the encoded metadata already provided in the transaction/material endpoint.

Additionally, since the decoded data is already available through another endpoint it is hard to justify adding this functionality.

**Comment by vae520283995** [comment]
> > I didn't use any package, I created the transaction according to the api of substrate-api-sidecar, so I hope the metadata returned by /transaction/material is the decoded data
> 
> Can you clarify why you need the decoded metadata? All the tooling I know of to make transactions can use the encoded metadata already provided in the transaction/material endpoint.
> 
As I said before, call_index is required in the transaction, and call_index comes from the decoded metadata

**Comment by vae520283995** [comment]
{
    "name":"Balances",
    "storage":Object{...},
    "calls":[
        {
            "name":"transfer",
            "args":Array[2],
            "documentation":Array[27]
        },
        Object{...},
        Object{...},
        {
            "name":"transfer_keep_alive",
            "args":Array[2],
            "documentation":Array[11]
        }
    ],
    "events":Array[8],
    "constants":Array[1],
    "errors":Array[8],
    "index":"4"
}
The above is part of the decoded metadata. For example, if I need to execute the Balances.transfer event, then I need to get the corresponding call_index, which can be obtained from the decoded metadata. The call_index is 0400

**Comment by emostov** [comment]
Yes, I completely understand that you need the call index to create the transaction. What I am saying though is that with most tooling you can extract the call index starting from the encoded metadata.

Can you show me the code you use to create a transaction? I might be able to help better that way.

**Comment by vae520283995** [comment]
> Can you show me the code you use to create a transaction? I might be able to help better that way.

I created the rawtransaction directly through the byte array,I did not use any other codes, starting from 0
for example:
tx = callindex + address_bytes + amount + 0x00 + nonce + 0x00 + sepc_version + transaction_version + genesishash + blockhash

**Comment by TarikGul** [comment]
@vae520283995 Is this something that is still an issue for you?

**Comment by vae520283995** [comment]
> @vae520283995 Is this something that is still an issue for you?

Still a issue
I want to add a query condition, like transaction/material?decodeMeta=true

**Comment by TarikGul** [comment]
Okay cool, I will bring this up with the team. I agree that a query param would be best suited here.
